src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -164,7 +167,21 @@ internal bool OnEdit(DocumentId documentId)
                 switch (projectReason)
                 {
                     case ProjectReadOnlyReason.MetadataNotAvailable:
-                        message = "Changes are not allowed if the project wasn't built when debugging started.";
+                        // TODO: Remove once https://github.com/dotnet/roslyn/issues/16657 is addressed
+                        bool deferredLoad = (_vsProject.ServiceProvider.GetService(typeof(SVsSolution)) as IVsSolution7)?.IsSolutionLoadDeferred() == true;
+                        if (deferredLoad)
+                        {
+                            message = "Changes are not allowed if the project wasn't loaded and built when debugging started." + Environment.NewLine + 
+                                      Environment.NewLine +
+                                      "'Lightweight solution load' is enabled for the current solution. " +
+                                      "Disable it to ensure that all projects are loaded when debugging starts.";
+
+                            s_encDebuggingSessionInfo?.LogReadOnlyEditAttemptedProjectNotBuiltOrLoaded();
+                        }
+                        else
+                        {
+                            message = "Changes are not allowed if the project wasn't built when debugging started.";
```

Comment:
Essentially if you have a missing dependency in project graph. You start debugging another project that loads the project thru Reflection.